src/cpp/ext/filters/census/client_filter.cc

```diff
@@ -44,15 +44,29 @@ constexpr uint32_t
 
 grpc_error_handle CensusClientCallData::Init(
     grpc_call_element* /* elem */, const grpc_call_element_args* args) {
-  auto tracer = args->arena->New<OpenCensusCallTracer>(args);
+  tracer_ = args->arena->New<OpenCensusCallTracer>(args);
   GPR_DEBUG_ASSERT(args->context[GRPC_CONTEXT_CALL_TRACER].value == nullptr);
-  args->context[GRPC_CONTEXT_CALL_TRACER].value = tracer;
+  args->context[GRPC_CONTEXT_CALL_TRACER].value = tracer_;
   args->context[GRPC_CONTEXT_CALL_TRACER].destroy = [](void* tracer) {
     (static_cast<OpenCensusCallTracer*>(tracer))->~OpenCensusCallTracer();
   };
   return GRPC_ERROR_NONE;
 }
 
+void CensusClientCallData::StartTransportStreamOpBatch(
+    grpc_call_element* elem, TransportStreamOpBatch* op) {
+  // Note that we are generating the overall call context here instead of in
+  // the constructor of `OpenCensusCallTracer` due to the semantics of
+  // `grpc_census_call_set_context` which allows the application to set the
+  // census context for a call anytime before the first call to
+  // `grpc_call_start_batch`.
+  if (!context_created_) {
```

Comment:
Instead of adding a `context_created_` data member, I think you can just trigger this when you see the `send_initial_metadata` op.  That will happen just once at the start of the call.

Reply:
I was thinking of doing that but this seems cleaner since we no longer need to make assumptions on the ordering of the ops.

Reply:
I think it's safe to assume that send_initial_metadata will always be the first op on the call; it can't proceed without that.  And this way, we are not unnecessarily increasing the per-call memory size (albeit by a small amount).